workflow-templates/im-build-db-ci.yml

```
@@ -0,0 +1,186 @@
+      # - name: Get db objects that have changed
+      #   id: changed-objects
+      #   shell: pwsh
+      #   run: |
+      #     $changedObjectsQuery = 
+      #       "SELECT
+      #           objectId,
+      #           schemaName,
+      #           objectName,
+      #           CASE objectType
+      #             WHEN 'U' THEN 'Tables'
+      #             WHEN 'V' THEN 'Views'
+      #             WHEN 'P' THEN 'StoredProcedures'
+      #             WHEN 'SO' THEN 'Sequences'
+      #             WHEN 'TF' THEN 'UserDefinedFunctions'
+      #             WHEN 'FN' THEN 'UserDefinedFunctions'
+      #             WHEN 'IF' THEN 'UserDefinedFunctions'
+      #             WHEN 'SN' THEN 'Synonyms'
+      #             ELSE objectType
+      #           END AS objectType,
+      #           operationType
+      #         FROM DBA.V_ChangedObjectsForSnapshot"
+              
+      #     $changedObjects = Invoke-Sqlcmd -ServerInstance $instanceName -Database $dbName -Query $changedObjectsQuery
+      #     $mappedObjects = $changedObjects | foreach-object { @{ schemaName=$_.schemaName; objectName=$_.objectName; objectType=$_.objectType; operationType=$_.operationType } }
+      #     $objectsAsJson = $mappedObjects | ConvertTo-Json -Compress
+
+      #     echo "::set-output name=json::$objectsAsJson"
+
+      # - name: Increment snapshots
+      #   uses: im-open/increment-database-object-snapshots@v1.0.0
+      #   with:
+      #     db-name: ${{ env.DB_NAME }}
+      #     instance-name: ${{ env.DB_SERVER_NAME }}
+      #     snapshot-path: ${{ env.SNAPSHOT_PATH }}
+      #     objects-to-increment: "${{ steps.changed-objects.outputs.json }}"
+      #     excluded-db-objects: ${{ env.EXCLUDED_DB_SNAPSHOT_OBJECTS }}
+
+      # - name: Setup git for snapshot commit
+      #   run: |
+      #     git config user.name github-actions
+      #     git config user.email github-actions@github.com
+
+      # TODO: Update the commit message if desired
+      # - name: Commit the snapshots
+      #   run: |
+      #     git add ${{ env.SNAPSHOT_PATH }}
+      #     git commit -m "Automated Snapshot Commit"
+      #     git push -f origin HEAD
+
+      
+      # Remove this if your build doesn't have need of publishing views for tables inside certain schemas
+      # - name: Create and publish views from tables
+      #   uses: im-open/sql-view-to-table-generator@v1.0.0
+      #   with:
+      #     schema-names: "dbo,CustomSchema" # TODO: Replace this with an actual list
+      #     db-name: ${{ env.DB_NAME }}
+      #     db-server: ${{ env.DB_SERVER_NAME }}
+      #     default-branch: ${{ env.DEFAULT_BRANCH }}
+      #     nuget-retrieval-url: "" # TODO: Replace this with the url for retrieving packages
+      #     publish-packages: true
+      #     nuget-publish-url: "" # TODO: Add the url for publishing packages
+      #     nuget-api-key: "${{ secrets.MY_GH_PACKAGES_ACCESS_TOKEN }}" # TODO: Update this with a real PAT
```

In the other workflows we've stayed away from adding placeholder or example values in the workflows.  We'd rather have it fail because it's missing than use a temp value that might work.
```suggestion
      #     nuget-api-key: '' # TODO: Update this with a real PAT
```